firmware/src/hal/sgx/src/trusted/secret_store.c

```diff
@@ -60,6 +66,56 @@ typedef struct {
     uint8_t* blob;
 } sealed_secret_t;
 
+/**
+ * @brief Adds a header to the given input data.
+ * The header is composed of the sha256 hash of the key.
+ * The resulting data is stored in the destination buffer.
+ *
+ * @param key The key to hash and set the header for.
+ * @param src The input data to prepend the key hash to.
+ * @param src_size The size of the input data.
+ * @param dest The destination buffer for generated data.
+ * @param dest_size The size of the destination buffer.
+ *
+ * @returns The number of bytes written to the destination buffer, or SEST_ERROR
+ * upon error.
+ */
+static size_t add_header(const char* key,
+                         const uint8_t* src,
+                         size_t src_size,
+                         uint8_t* dest,
+                         size_t dest_size) {
+    if (dest_size < HASH_LENGTH + src_size) {
+        LOG("Failed to add header - destination buffer is too small\n");
+        return SEST_ERROR;
+    }
+
+    sha256_init(&G_sha256_ctx);
+    sha256_update(&G_sha256_ctx, (const uint8_t*)key, strlen(key));
+    sha256_final(&G_sha256_ctx, dest);
+
+    if (src_size)
+        platform_memmove(dest + HASH_LENGTH, src, src_size);
+
+    return HASH_LENGTH + src_size;
+}
+
+/**
+ * @brief Checks if the data buffer contains the correct header for the given
+ * key.
+ *
+ * @param key The key to validate the header against.
+ * @param data The buffer containing the data to validate.
+ *
+ * @returns true if the header is valid, false otherwise.
+ */
+static bool is_header_valid(const char* key, const uint8_t* data) {
+    uint8_t expected_header[HASH_LENGTH];
+    add_header(key, NULL, 0, expected_header, HASH_LENGTH);
+
+    return memcmp(expected_header, data, HASH_LENGTH) == 0;
```

Comment:
We need to make sure the comparison doesn't read out of bounds.

Reply:
Done! See last commit